sdk/provisioning/Azure.Provisioning/src/ModuleConstruct.cs

```diff
@@ -182,18 +182,36 @@ internal void WriteOutputs(MemoryStream stream)
             GetAllOutputsRecursive(this, outputsToWrite, false);
             foreach (var output in outputsToWrite)
             {
-                string value;
-                if (output.IsLiteral || ReferenceEquals(this, output.Resource.ModuleScope))
-                {
-                    value = output.IsLiteral ? $"'{output.Value}'" : output.Value;
-                }
-                else
+                stream.WriteLine(GetOutputString(output));
+            }
+        }
+
+        private string GetOutputString(Output output)
+        {
+            string value;
+
+            if (output.IsLiteral)
+            {
+                value = output.Kind switch
                 {
-                    value = $"{output.Resource.ModuleScope!.Name}.outputs.{output.Name}";
-                }
-                string name = output.Name;
-                stream.WriteLine($"output {name} string = {value}");
+                    BicepKind.Bool => output.Value.ToLower(),
```

Comment:
Can `Value` be `null` and if so should we coerce to `false`?

Reply:
No the value can't be null.